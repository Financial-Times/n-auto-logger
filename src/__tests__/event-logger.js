import logger from '../index';
import createEventLogger from '../event-logger';

jest.mock('@financial-times/n-logger');

describe('createEventLogger', () => {
	afterEach(() => {
		jest.resetAllMocks();
	});

	const commonMeta = {
		operation: 'test',
		action: '',
		userId: 'test',
		transactionId: 'test',
		a: 'test',
		b: 'test',
	};

	const commonEvent = createEventLogger(commonMeta);

	describe('fires logger', () => {
		it('info with .start()', () => {
			commonEvent.start();
			expect(logger.info.mock.calls).toMatchSnapshot();
		});

		it('info with .success()', () => {
			commonEvent.success({ d: 'some data', e: 'some other data' });
			expect(logger.info.mock.calls).toMatchSnapshot();
		});

		it('failureLogger with .failure()', () => {
			commonEvent.failure({ message: 'some error message' });
			expect(logger.error.mock.calls).toMatchSnapshot();
		});
	});

	describe('filter in input meta data of', () => {
		it('empty or undefined field', () => {
			commonEvent.start();
			expect(logger.info.mock.calls).toMatchSnapshot();
		});

		it('fields defined in LOGGER_MUTE_FIELDS', () => {
			// currently ENV_VAR would only be effective when they were set before createEventLogger
			process.env.LOGGER_MUTE_FIELDS = 'transactionId, userId';
			const event = createEventLogger(commonMeta);
			event.start();
			expect(logger.info.mock.calls).toMatchSnapshot();
			delete process.env.LOGGER_MUTE_FIELDS;
		});

		it('data nested in user field', () => {
			const commonMetaWithUser = {
				...commonMeta,
				user: { message: 'please ask help center' },
			};
			const event = createEventLogger(commonMetaWithUser);
			event.start();
			expect(logger.info.mock.calls).toMatchSnapshot();
		});

		it('function fields', () => {
			const commonMetaWithFunc = {
				...commonMeta,
				func: () => null,
			};
			const event = createEventLogger(commonMetaWithFunc);
			event.start();
			expect(logger.info.mock.calls).toMatchSnapshot();
		});
	});
});
