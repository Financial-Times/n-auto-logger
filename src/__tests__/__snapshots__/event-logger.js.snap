// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createEventLogger filter in input meta data of data nested in user field 1`] = `
Array [
  Array [
    Object {
      "a": "test",
      "b": "test",
      "operation": "test",
      "transactionId": "test",
      "userId": "test",
    },
  ],
]
`;

exports[`createEventLogger filter in input meta data of empty or undefined field 1`] = `
Array [
  Array [
    Object {
      "a": "test",
      "b": "test",
      "operation": "test",
      "transactionId": "test",
      "userId": "test",
    },
  ],
]
`;

exports[`createEventLogger filter in input meta data of fields defined in LOGGER_MUTE_FIELDS 1`] = `
Array [
  Array [
    Object {
      "a": "test",
      "b": "test",
      "operation": "test",
    },
  ],
]
`;

exports[`createEventLogger filter in input meta data of function fields 1`] = `
Array [
  Array [
    Object {
      "a": "test",
      "b": "test",
      "operation": "test",
      "transactionId": "test",
      "userId": "test",
    },
  ],
]
`;

exports[`createEventLogger fires logger failureLogger with .failure() 1`] = `
Array [
  Array [
    Object {
      "a": "test",
      "b": "test",
      "category": "CUSTOM_ERROR",
      "message": "some error message",
      "operation": "test",
      "result": "failure",
      "transactionId": "test",
      "userId": "test",
    },
  ],
]
`;

exports[`createEventLogger fires logger info with .start() 1`] = `
Array [
  Array [
    Object {
      "a": "test",
      "b": "test",
      "operation": "test",
      "transactionId": "test",
      "userId": "test",
    },
  ],
]
`;

exports[`createEventLogger fires logger info with .success() 1`] = `
Array [
  Array [
    Object {
      "a": "test",
      "b": "test",
      "data": Object {
        "d": "some data",
        "e": "some other data",
      },
      "operation": "test",
      "result": "success",
      "transactionId": "test",
      "userId": "test",
    },
  ],
]
`;
